Commands/InteractionCommands/ClearInteractions.cs

```diff
@@ -0,0 +1,264 @@
+                    }
+                    catch (DSharpPlus.Exceptions.NotFoundException)
+                    {
+                        // User is not in the server, assume not mod
+                        continue;
+                    }
+
+                    if (GetPermLevel(member) >= ServerPermLevel.TrialModerator)
+                    {
+                        messagesToClear.Remove(message);
+                    }
+                }
+            }
+
+            // Match text
+            if (match != "")
+            {
+                foreach (var message in messagesToClear.ToList())
+                {
+                    if (!message.Content.ToLower().Contains(match.ToLower()))
+                    {
+                        messagesToClear.Remove(message);
+                    }
+                }
+            }
+
+            // Bots only
+            if (botsOnly)
+            {
+                if (humansOnly)
+                {
+                    await ctx.FollowUpAsync(new DiscordFollowupMessageBuilder().WithContent($"{Program.cfgjson.Emoji.Error} You can't use `bots_only` and `humans_only` together! Pick one or the other please.").AsEphemeral(true));
+                    return;
+                }
+
+                foreach (var message in messagesToClear.ToList())
+                {
+                    if (!message.Author.IsBot)
+                    {
+                        messagesToClear.Remove(message);
+                    }
+                }
+            }
+
+            // Humans only
+            if (humansOnly)
+            {
+                foreach (var message in messagesToClear.ToList())
+                {
+                    if (message.Author.IsBot)
+                    {
+                        messagesToClear.Remove(message);
+                    }
+                }
+            }
+
+            // Images only
+            if (imagesOnly)
+            {
+                if (linksOnly)
+                {
+                    await ctx.FollowUpAsync(new DiscordFollowupMessageBuilder().WithContent($"{Program.cfgjson.Emoji.Error} You can't use `images_only` and `links_only` together! Pick one or the other please.").AsEphemeral(true));
+                    return;
+                }
+
+                foreach (var message in messagesToClear.ToList())
+                {
+                    if (message.Attachments.Count == 0)
+                    {
+                        messagesToClear.Remove(message);
+                    }
+                }
+            }
+
+            // Links only
+            if (linksOnly)
+            {
+                foreach (var message in messagesToClear.ToList())
+                {
+                    if (!Constants.RegexConstants.url_rx.IsMatch(message.Content.ToLower()))
+                    {
+                        messagesToClear.Remove(message);
+                    }
+                }
+            }
+
+            bool skipped = false;
```

Comment:
I like your way of commenting the logic behind things, why stop here
```suggestion
           // Skip messages older than 2 weeks, since Discord won't let us delete them anyway

            bool skipped = false;
```